azure-spring-boot/src/test/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelperUnitTest.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.spring;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.EnvironmentCapable;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@TestPropertySource(locations = "classpath:application.properties")
+public class KeyVaultEnvironmentPostProcessorHelperUnitTest {
+
+    @Autowired
+    ApplicationContext context;
+
+    private KeyVaultEnvironmentPostProcessorHelper keyVaultEnvironmentPostProcessorHelper;
+    private ConfigurableEnvironment environment;
+
+    @Before
+    public void setup() {
+        environment = (ConfigurableEnvironment) context.getEnvironment();
+    }
+
+    @Test
+    public void testMSIAuthPropertiesNotInitialized() { 
+        assertFalse("Environment should not contain MSI_ENDPOINT when not on Azure App Service environment",
+        environment.containsProperty("MSI_ENDPOINT"));
+
+        assertFalse("Environment should not contain MSI_SECRET when not on Azure App Service environment",
+        environment.containsProperty("MSI_SECRET"));
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testMSIAuthentication() {
```

Comment:
I can't really see what this test verifies except that it except a RuntimeException to be thrown?

Reply:
Agreed, thanks, cc @jalanubha